gtk4/Cargo.toml

```diff
@@ -22,7 +22,7 @@ name = "gtk4"
 [features]
 purge-lgpl-docs = ["gtk-rs-lgpl-docs", "gdk/purge-lgpl-docs", "gsk/purge-lgpl-docs"]
 embed-lgpl-docs = ["gtk-rs-lgpl-docs", "gdk/embed-lgpl-docs", "gsk/embed-lgpl-docs"]
-dox = ["gdk/dox", "gsk/dox", "ffi/dox"]
+dox = ["gdk/dox", "gsk/dox", "ffi/dox", "x11", "wayland"]
```

Comment:
I'm not sure whether this is right to do. But then we don't have any other way to do it, if we want to show the possible feature there's.

Reply:
Will this still build on a system without x11/Wayland then? Probably not?

Reply:
Normally we don't link when the `dox` feature is enabled, so it should be fine. Is the CI checking if `cargo doc` works with the `dox` feature? If so we should be fine.

Reply:
> Will this still build on a system without x11/Wayland then? Probably not?

If you're building docs locally? I can't try on those platforms, I suppose it should. Maybe we can add another cfg to only enable them under linux as those apis are not useful for macOS/Windows.

> Is the CI checking if cargo doc works with the dox feature? If so we should be fine.

Yes, the CI does build the docs, it just only publish them if the job was running on the master branch.

Reply:
I have added an extra cfg to ensure it's only set for linux. Although, it seems rustdoc doesn't display the configured feature attribute for the re-exports. 

Anyway, this seems to be working fine locally